src/zfs-utils/zfs-utils.initcpio.hook

```diff
@@ -25,6 +25,27 @@ zfs_get_bootfs () {
     return 1
 }
 
+zfs_decrypt_fs() {
+    dataset=$1
+    
+    # check if 'zfs load-key' is available
+    zfs 2>&1 | grep load-key > /dev/null || return 0
```

Comment:
This will check if encryption is available, so this change should also work on stable